src/main/sensors/gyro.c

```diff
@@ -510,3 +516,39 @@ bool gyroSyncCheckUpdate(void)
 
     return gyroDev[0].intStatusFn(&gyroDev[0]);
 }
+
+static float dynLpfCutoffFreq(float throttle, uint16_t dynLpfMin, uint16_t dynLpfMax, uint8_t expo) {
+    const float expof = expo / 10.0f;
+    static float curve;
+    curve = throttle * (1 - throttle) * expof + throttle;
+    return (dynLpfMax - dynLpfMin) * curve + dynLpfMin;
+}
+
+static float dynThrottle(float throttle) {
+    return throttle * (1 - (throttle * throttle) / 3.0f) * 1.5f;
+}
+
+void gyroUpdateDynamicLpf(void) {
+    if (!gyroConfig()->useDynamicLpf) {
+        return;
+    }
+
+    const float throttle = scaleRangef((float) rcCommand[THROTTLE], getThrottleIdleValue(), motorConfig()->maxthrottle, 0.0f, 1.0f);
+
+    uint16_t cutoffFreq;
+    if (gyroConfig()->gyroDynamicLpFCurveExpo > 0) {
+        cutoffFreq = dynLpfCutoffFreq(throttle, gyroConfig()->gyroDynamicLpfMinHz, gyroConfig()->gyroDynamicLpfMaxHz, gyroConfig()->gyroDynamicLpFCurveExpo);
+    } else {
+        cutoffFreq = fmax(dynThrottle(throttle) * gyroConfig()->gyroDynamicLpfMaxHz, gyroConfig()->gyroDynamicLpfMinHz);
+    }
```

Comment:
I suggest to use only the new curve with `1-10` setting

Reply:
I was wondering if to port the legacy one. Now I know. Thanks for the hint!